common/src/main/java/com/wynntils/wc/utils/ChatItemUtils.java

```diff
@@ -271,29 +272,29 @@ private static Component createItemComponent(GearItemStack item) {
     }
 
     private static String encodeString(String text) {
-        String encoded = "";
+        StringBuilder encoded = new StringBuilder();
         for (char c : text.toCharArray()) {
             int value = c - 32; // offset by 32 to ignore ascii control characters
-            encoded += new String(Character.toChars(value + OFFSET)); // get encoded representation
+            encoded.append(new String(Character.toChars(value + OFFSET))); // get encoded representation
         }
-        return encoded;
+        return encoded.toString();
     }
 
     private static String encodeNumber(int value) {
         return new String(Character.toChars(value + OFFSET));
     }
 
     private static String decodeString(String text) {
-        String decoded = "";
+        StringBuilder decoded = new StringBuilder();
         for (int i = 0; i < text.length(); i += 2) {
             int value = text.codePointAt(i) - OFFSET + 32;
-            decoded += (char) value;
+            decoded.append((char) value);
         }
-        return decoded;
+        return decoded.toString();
```

Comment:
Actually modern Hotspot is very good at rewriting string operations to use StringBuilder under the hood, so in most cases you can keep using the easy-to-read syntax of string manipulation, and get StringBuilder performance for free.